Source/plugins/IShell.h

```diff
@@ -255,66 +273,82 @@ namespace PluginHost {
         virtual Core::hresult Hibernate(const uint32_t timeout) = 0;
         virtual reason Reason() const = 0;
 
+        virtual void Register(IConnectionServer::INotification* sink) = 0;
```

Comment:
Do we still need these public? I can, I assume, QueryInterface<IConnectionServer>() and than Register/Unregister? Or am I missing something here? This would be a conveenince method I guess but than I have to implement it on all IShells ?